ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp

```diff
@@ -86,24 +86,27 @@ namespace ngraph
 
                 const auto mask_size = mask.size();
 
-                std::copy(input, input + shape_size(input_shape), output);
-
                 int num_regions = 0;
                 int end_index = 0;
+                int output_size = 0;
 
                 if (do_softmax)
                 {
                     // Region layer (Yolo v2)
                     num_regions = regions;
                     end_index = width * height;
+                    output_size = shape_size(input_shape);
                 }
                 else
                 {
                     // Yolo layer (Yolo v3)
                     num_regions = mask_size;
                     end_index = width * height * (classes + 1);
+                    output_size = width * height * num_regions * (classes + coords + 1);
                 }
 
+                std::copy(input, input + output_size, output);
```

Comment:
Do we need to add more tests for nGraph reference implementation?

Reply:
Are tests for the reference implementation located in test/backend?
As for more tests, I could templatize them and basically replicate parameters from CPU tests, is it ok?

Reply:
I propose to extend current backend tests in order to cover the fix in the nGraph unit-tests.

Reply:
Added test, which reproduce the issue (fails on master)